### PR TITLE
Fix job generator with the new parameter filter in configuration

### PR DESCRIPTION
**Pull request by momoss**
Fix job generator with the new parameter filters:

```
configuration:
      filters:
           data: {}
           structure:
                 scope: ecommerce
```

| Q | A |
| --- | --- |
| Added Specs | N |
| Generate fixtures and install | Y |
| Generate products and import | Y |
| Review and 2 GTM | Y |
|  Tech Doc | N |

### CODE REVIEW

File: DependencyInjection/Configuration/FixtureGeneratorConfiguration.php [hunk 1]
```diff
@@ -122,7 +122,33 @@ public function getConfigTreeBuilder()
...
+                                        ->children()
+                                            ->booleanNode('uploadAllowed')->end()
+                                            ->scalarNode('delimiter')->end()
+                                            ->scalarNode('enclosure')->end()
+                                            ->scalarNode('escape')->end()
+                                            ->booleanNode('enabled')->end()
+                                            ->scalarNode('categoriesColumn')->end()
+                                            ->scalarNode('familyColumn')->end()
+                                            ->scalarNode('groupsColumn')->end()
+                                            ->scalarNode('withHeader')->end()
```
Comment (SamirBoulil) [inline DependencyInjection/Configuration/FixtureGeneratorConfiguration.php]: BooleanNode ?

File: DependencyInjection/Configuration/FixtureGeneratorConfiguration.php [hunk 2]
```diff
@@ -122,7 +122,33 @@ public function getConfigTreeBuilder()
...
+                                            ->scalarNode('filePath')->end()
+                                            ->scalarNode('multiple')->end()
```
Comment (SamirBoulil) [inline DependencyInjection/Configuration/FixtureGeneratorConfiguration.php]: Boolean ?

File: Generator/JobGenerator.php [hunk 3]
```diff
@@ -83,7 +83,10 @@ protected function generateJob($code, array $jobConfig)
         $job->setJobName($jobConfig['alias']);
         $job->setLabel($jobConfig['label']);
         $job->setType($jobConfig['type']);
-        $job->setRawParameters($jobConfig['configuration']);
+
+        if (isset($jobConfig['configuration']) && null !== $jobConfig['configuration']) {
```
Comment (SamirBoulil) [inline Generator/JobGenerator.php]: isset(null) = false I beliefe remove second part ?

**Comment by SamirBoulil** [comment]
Small fix then GTM, nice one ;)

File: DependencyInjection/Configuration/FixtureGeneratorConfiguration.php (hunk 1 above)
Comment (momoss) [inline DependencyInjection/Configuration/FixtureGeneratorConfiguration.php]: Exact bro,I'll fix all, thank's

File: DependencyInjection/Configuration/FixtureGeneratorConfiguration.php [hunk 4]
```diff
@@ -122,7 +122,33 @@ public function getConfigTreeBuilder()
...
+                                            ->scalarNode('enclosure')->end()
+                                            ->scalarNode('escape')->end()
+                                            ->booleanNode('enabled')->end()
+                                            ->scalarNode('categoriesColumn')->end()
+                                            ->scalarNode('familyColumn')->end()
+                                            ->scalarNode('groupsColumn')->end()
+                                            ->booleanNode('withHeader')->end()
+                                            ->scalarNode('filePath')->end()
+                                            ->booleanNode('multiple')->end()
+                                            ->scalarNode('header')->end()
```
Comment (pierallard) [inline DependencyInjection/Configuration/FixtureGeneratorConfiguration.php:137]: boolean

**Comment by pierallard** [comment]
Check CE fixtures jobs

**Comment by pierallard** [comment]
Add default filters for product export

File: DependencyInjection/Configuration/FixtureGeneratorConfiguration.php (hunk 4 above)
Comment (momoss) [inline DependencyInjection/Configuration/FixtureGeneratorConfiguration.php:137]: No, it's a scalar node

**Comment by pierallard** [comment]
GTM

**Pull request merged**